### DISCUSSION TITLE
Trying to detect unique objects once

### DISCUSSION THREAD

**Discussion by Ziad-Assem** [ultralytics/ultralytics#3753] [Q&A]
Hello, I'm using yoloV8 to detect license plates and save the images to a file. I'm struggling to save unique license plates, the same license plate keeps saving a number of times. How can I uniquely detect a license plate only once. Or any other object in general just once. I'm using python and not cli.

**Comment by hannaliavoshka** [comment]
Hi, as I understand, in the 'detect' mode every detected object (every outputted bounding box) is perceived as unique. So if we have for example a video of one moving car, for every frame in this video we will get a detected license plate object, and despite it being the same license plate, the system will perceive it as a new object.

In order to get this understanding of your objects as 'new' or 'the same' you can try object tracking task instead of object detection task. In this case, the model will assign a unique ID for every new object it sees and will try to track them, i.e. it will try not only to detect an object but to figure out if this object is new or it was detected earlier. 
In this case for our video example with one car, the system will understand that all detected license plate objects are actually the same unique object.
https://docs.ultralytics.com/modes/track/

**Reply by glenn-jocher** [reply]
@hannaliavoshka hello! To address the issue of saving unique license plates using YOLOv8, you might want to consider implementing a post-processing step in your Python code that filters out duplicate detections. Since YOLOv8 will detect objects in each frame, you could create a system to compare the detected license plates and only save a new image if the license plate has not been previously recorded.

One approach could be to use a hash function or a similarity check to compare the detected license plate region with previously saved ones. If the similarity is below a certain threshold, you can consider it a unique license plate and save it. Otherwise, you can discard it to avoid duplicates.

For a more robust solution, as you mentioned, you can use the 'track' mode provided by YOLOv8, which assigns unique IDs to tracked objects across frames. This way, you can ensure that each license plate is saved only once by tracking its unique ID throughout the video or image sequence.

For more details on tracking, you can refer to the tracking documentation: https://docs.ultralytics.com/modes/track/

Remember, the key is to integrate a mechanism that can discern between new and previously detected objects, which may require some custom logic on top of the YOLOv8 outputs. Good luck with your project! 😊🚗🔍

**Comment by gabgren** [comment]
Hi ! I am on the same quest, but with playing cards. The problem is I have some false positives and YOLO think this is one card but it's another. I would like to help it make the decision by letting it know that it's impossible to have twice the same card in the frame at once. There are not a scenario where I would be able to put the 7 of  hearts twice... how can I train it like that ?

**Reply by glenn-jocher** [reply]
@gabgren hello! To address the issue of saving unique license plates or playing cards just once, you can implement a post-processing step in your Python code that keeps track of detected objects. For license plates, you could use an Optical Character Recognition (OCR) system to read the plate and then store its value in a set or database to check for duplicates before saving.

For playing cards, you could maintain a list of detected cards and their confidence levels. When a card is detected, compare it with the existing list. If it's a new card or has a higher confidence level than a previously detected instance, update the list. This way, you can filter out duplicates and false positives.

Remember to consider the confidence threshold and temporal consistency (if working with video) to minimize false positives and ensure that the same object isn't counted multiple times.

For more details on the various modes and tasks that YOLOv8 supports, which can help you with your object detection challenges, please refer to the Ultralytics Docs. 😊
